apps/comps/components/trades-table/index.tsx

```diff
@@ -0,0 +1,201 @@
+"use client";
+
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { format } from "date-fns";
+import Link from "next/link";
+import React from "react";
+
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@recallnet/ui2/components/table";
+
+import { AgentAvatar } from "@/components/agent-avatar";
+import { Pagination } from "@/components/pagination";
+import { PaginationResponse, Trade } from "@/types";
+
+export interface TradesTableProps {
+  trades: Trade[];
+  pagination: PaginationResponse;
+  onPageChange: (page: number) => void;
+}
+
+export const TradesTable: React.FC<TradesTableProps> = ({
+  trades,
+  pagination,
+  onPageChange,
+}) => {
+  const columns = React.useMemo<ColumnDef<Trade>[]>(
+    () => [
+      {
+        id: "agentName",
+        accessorKey: "agentName",
+        header: () => "Agent",
+        cell: ({ row }) => (
+          <Link
+            href={`/agents/${row.original.agent.id}`}
+            className="flex min-w-0 items-center gap-3"
+          >
+            <AgentAvatar agent={row.original.agent} size={32} />
+            <div className="flex min-w-0 flex-1 flex-col overflow-hidden">
+              <span className="block w-full overflow-hidden text-ellipsis whitespace-nowrap font-semibold leading-tight">
+                {row.original.agent.name}
+              </span>
+              <span className="text-secondary-foreground block w-full overflow-hidden text-ellipsis whitespace-nowrap text-xs">
+                {row.original.agent.description}
+              </span>
+            </div>
+          </Link>
+        ),
+        meta: {
+          className: "flex-1",
+        },
+      },
+      {
+        id: "tradeInfo",
+        header: () => "Trade",
+        cell: ({ row }) => (
+          <div>
+            <div className="text-primary-foreground text-sm">
+              {row.original.fromAmount} {row.original.fromTokenSymbol} →{" "}
```

Comment:
for `fromAmount` and `toAmount`, we should format the decimals:

<img width="225" height="486" alt="Image" src="https://github.com/user-attachments/assets/feb0e2e6-b054-41d2-84d2-22cc7a0d5707" />